quickjs.c

```diff
@@ -537,6 +538,73 @@ typedef enum JSFunctionKindEnum {
     JS_FUNC_ASYNC_GENERATOR = (JS_FUNC_GENERATOR | JS_FUNC_ASYNC),
 } JSFunctionKindEnum;
 
+typedef struct InlineCacheRingItem {
+    JSShape* shape;
+    uint32_t prop_offset;
+} InlineCacheRingItem;
+
+typedef struct InlineCacheRingSlot {
+    JSAtom atom;
+    InlineCacheRingItem buffer[IC_CACHE_ITEM_CAPACITY];
+    uint8_t index;
+} InlineCacheRingSlot;
+
+typedef struct InlineCacheHashSlot {
+    JSAtom atom;
+    uint32_t index;
+    struct InlineCacheHashSlot *next;
+} InlineCacheHashSlot;
+
+typedef struct InlineCache {
+    uint32_t count;
+    uint32_t capacity;
+    uint32_t hash_bits;
+    JSContext* ctx;
```

Comment:
I think you can get rid of this and pass the JSContext everywhere explicitly?

Reply:
Done